src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java

```diff
@@ -0,0 +1,316 @@
+import org.opensearch.security.tools.Hasher;
+
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_MIN_LENGTH;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX;
+import static org.opensearch.security.user.UserService.generatePassword;
+
+/**
+ * This class updates the security related configuration, as needed.
+ */
+public class SecuritySettingsConfigurer extends Installer {
+
+    /**
+     * Configures security related changes to the opensearch configuration
+     * 1. Checks if plugins is already configuration. If yes, exit
+     * 2. Sets the custom admin password (Generates one if none is provided)
+     * 3. Write the security config to opensearch.yml
+     */
+    public void configureSecuritySettings() {
+        checkIfSecurityPluginIsAlreadyConfigured();
+        updateAdminPassword();
+        writeSecurityConfigToOpenSearchYML();
+    }
+
+    /**
+     * Replaces the admin password in internal_users.yml with the custom or generated password
+     */
+    static void updateAdminPassword() {
+        String ADMIN_PASSWORD = "";
+        String initialAdminPassword = System.getenv("initialAdminPassword");
+        String ADMIN_PASSWORD_FILE_PATH = OPENSEARCH_CONF_DIR + "initialAdminPassword.txt";
+        String INTERNAL_USERS_FILE_PATH = OPENSEARCH_CONF_DIR + "opensearch-security" + File.separator + "internal_users.yml";
+        boolean shouldValidatePassword = environment.equals(ExecutionEnvironment.demo);
+        try {
+            final PasswordValidator passwordValidator = PasswordValidator.of(
+                Settings.builder()
+                    .put(SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, "(?=.*[A-Z])(?=.*[^a-zA-Z\\\\d])(?=.*[0-9])(?=.*[a-z]).{8,}")
+                    .put(SECURITY_RESTAPI_PASSWORD_MIN_LENGTH, 8)
+                    .build()
+            );
+
+            // Read custom password
+            if (initialAdminPassword != null && !initialAdminPassword.isEmpty()) {
+                ADMIN_PASSWORD = initialAdminPassword;
+            } else {
+                File adminPasswordFile = new File(ADMIN_PASSWORD_FILE_PATH);
+                if (adminPasswordFile.exists() && adminPasswordFile.length() > 0) {
+                    try (BufferedReader br = new BufferedReader(new FileReader(ADMIN_PASSWORD_FILE_PATH, StandardCharsets.UTF_8))) {
+                        ADMIN_PASSWORD = br.readLine();
+                    }
+                }
+            }
+
+            // If script execution environment is set to demo, validate custom password, else if set to test, skip validation
+            if (shouldValidatePassword
+                && !ADMIN_PASSWORD.isEmpty()
+                && passwordValidator.validate("admin", ADMIN_PASSWORD) != RequestContentValidator.ValidationError.NONE) {
+                System.out.println("Password " + ADMIN_PASSWORD + " is weak. Please re-try with a stronger password.");
```

Comment:
Nit: we should also list out the rules for a stronger password (e.g. at least 1 symbol...etc.).

Reply:
Was there a reason we couldn't include the validation failure enum message here directly?

From (https://github.com/opensearch-project/security/pull/3057/files#diff-4b39b8c38e1e191dbba2db25c2f52a7588c6b6087a5d0ff2795d97bd0b2bc776), it looks like the validation rules from `RequestContentValidator.ValidationError` can provide some details on why the password was "bad" (e.g. too short, too similar to username, too weak per zxcvbn checks) and that info could be helpful to the users. 

Will make PR to add some more context to address #4048 anyways.